tutorials/calibration_flow/static_quant.py

```diff
@@ -96,8 +121,33 @@ def from_observed(cls, observed_linear):
         quantized_linear = cls(observed_linear.in_features, observed_linear.out_features, observed_linear.act_obs, observed_linear.weight_obs, observed_linear.weight, observed_linear.bias)
         return quantized_linear
 
-def apply_static_quant2(observed_linear):
-    return QuantizedLinear.from_observed(observed_linear)
+class QuantizedLinearFloatX(torch.nn.Module):
+    def __init__(self, in_features: int, out_features: int, act_obs: torch.nn.Module, weight_obs: torch.nn.Module, weight: torch.Tensor, bias: torch.Tensor):
+        super().__init__()
+        self.act_scale, self.act_zero_point = act_obs.calculate_qparams()
+        weight_scale, weight_zero_point = weight_obs.calculate_qparams()
+        assert weight.dim() == 2
+        block_size = (1, weight.shape[1])
+        target_dtype = torch.float8_e4m3fn
+        self.qweight = to_affine_quantized_floatx_static(weight, weight_scale, block_size, target_dtype, Float8LayoutType(mm_config=None))
```

Comment:
can you use a branch that's based on target_dtype? I think it would cleaner if we just expose a single static quant API for intx/floatx in the end

Reply:
Done